File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int lft=0;
        int rght = heights.size()-1;
        int ans=0;
        while(lft<=rght){
            int area = min(heights[lft],heights[rght])*(rght-lft);
            ans = max(ans,area); 
            if(heights[lft]<heights[rght]){lft++;}
            else{rght--;}}return ans;
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        n =len(height)
        l = 0
        r = n-1
        maxar = 0
        while(l<r):
            wid = r-l
            heig = min(height[l],height[r])
            maxar = max(wid*heig,maxar)
            if height[l]>height[r]:
                r-=1
            else :
                l+=1
        return maxar